src/gui/UserStatusSelector.qml

```diff
@@ -19,103 +19,182 @@ import QtQuick.Controls 2.15
 import QtQuick.Window 2.15
 import com.nextcloud.desktopclient 1.0 as NC
 
+import Style 1.0
+
 ColumnLayout {
     id: rootLayout
     spacing: 0
     property NC.UserStatusSelectorModel userStatusSelectorModel
 
     Label {
-        Layout.topMargin: 16
-        Layout.leftMargin: 8
-        Layout.rightMargin: 8
-        Layout.bottomMargin: 8
+        Layout.topMargin: Style.standardSpacing * 2
+        Layout.leftMargin: Style.standardSpacing
+        Layout.rightMargin: Style.standardSpacing
+        Layout.bottomMargin: Style.standardSpacing
         Layout.alignment: Qt.AlignTop | Qt.AlignHCenter
         font.bold: true
         text: qsTr("Online status")
+        color: Style.ncTextColor
     }
         
     GridLayout {
-        Layout.margins: 8
+        id: topButtonsLayout
+
+        Layout.margins: Style.standardSpacing
         Layout.alignment: Qt.AlignTop
         columns: 2
         rows: 2
-        columnSpacing: 8
-        rowSpacing: 8
+        columnSpacing: Style.standardSpacing
+        rowSpacing: Style.standardSpacing
 
-        Button {
-            Layout.fillWidth: true
+        property int maxButtonHeight: 0
+        function updateMaxButtonHeight(newHeight) {
+            maxButtonHeight = Math.max(maxButtonHeight, newHeight)
+        }
```

Comment:
if I understand it correctly you are trying to get all buttons to expand vertically to occupy the same height
could you not achieve that by using `Layout.fillHeight` ?

Reply:
No, as `Layout.fillHeight` only expands the height on a per-row basis, and since only the bottom two buttons have two lines of text, the top row of buttons will be shorter than the bottom row of buttons. This ensures all buttons are the same size regardless of row